Add Vagrantfile and subman-devel ansible role

kahowell opened the pull request:
See README for more information.

It should be possible to use the subman-devel role to configure one's
own host for development work as well, though that can be handled
outside of this commit.

jirihnidek commented:
When `README` file starts to grow up now, what about renaming it to `README.md` and use Markdown for formatting?

When you type e.g. `sudo subscription-manager version`, then it reports missing `GTK_VERSION`:

    /vagrant/src/subscription_manager/ga_loader.py:240: UserWarning: GTK_VERSION is unset in version.py.  Using GTK 3
      warnings.warn("GTK_VERSION is unset in version.py.  Using GTK %s" % gtk_version_from_build)

I suggest to fix this using `export SUBMAN_GTK_VERSION=3` for `centos7`  and `fedora25` and `export SUBMAN_GTK_VERSION=2` for `fedora6`.

kahowell commented:
0f075c4..7e028d7:
 - move `roles`, `plugins`, and `vagrant.yml` to vagrant subfolder
 - set `SUBMAN_GTK_VERSION`
 - change `README` to `README.md`, few tweaks to README

@cnsnyder, @jirihnidek, I think that addresses all your feedback. Also, I did try the dbus interfaces on both centos6 and centos7 (using `dbus-send` and `busctl`). I didn't encounter any issues.

kahowell commented:
7e028d7..07370d5:
 - remove unused `subman_gtk_version` variable
 - fix `README` url formatting
